### PR TITLE
build(deps): bump to typespec 0.61.0

### PR DESCRIPTION
**Pull request by archerzz**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/package.json [hunk 1]
```diff
@@ -38,23 +38,23 @@
     "@azure-tools/cadl-ranch-api": "workspace:~"
   },
   "devDependencies": {
-    "@azure-tools/typespec-autorest": "0.46.0",
-    "@azure-tools/typespec-azure-resource-manager": "0.46.0",
-    "@azure-tools/typespec-client-generator-core": "0.46.0",
+    "@azure-tools/typespec-autorest": "0.47.0",
+    "@azure-tools/typespec-azure-resource-manager": "0.47.0",
+    "@azure-tools/typespec-client-generator-core": "0.47.1",
```
Comment (haolingdong-msft) [inline packages/cadl-ranch-specs/package.json:43]: wrong version? Should it be "0.47.0"?

Comment (archerzz) [reply packages/cadl-ranch-specs/package.json:43]: It's `0.47.1`: https://www.npmjs.com/package/@azure-tools/typespec-client-generator-core/v/0.47.1

Comment (haolingdong-msft) [reply packages/cadl-ranch-specs/package.json:43]: Ah, a bit strange, curious why does TCGC release another version with in several hours. /cc @tadelesh

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/package.json:43]: I think Chenjie mentioned in chat that there is a workaround (broken by new compiler).
You may need to update it in our repo too.

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/cadl-ranch-specs/package.json (hunk 1 above)
Comment (haolingdong-msft) [reply packages/cadl-ranch-specs/package.json:43]: Oh, seems I did not find that chat. Will update for Java.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**